pkg/services/account/workspace/audit.go

```diff
@@ -0,0 +1,133 @@
+package workspace
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/de-tools/data-atlas/pkg/models/domain"
+)
+
+// GetDLTAudit computes DLT pipelines audit for a workspace over the period.
+func GetDLTAudit(
+	ctx context.Context,
+	ws domain.Workspace,
+	startTime, endTime time.Time,
+	costManager CostManager,
+) (domain.AuditReport, error) {
+	// Collect DLT-related usage
+	resources := domain.WorkspaceResources{WorkspaceName: ws.Name, Resources: []string{"dlt_pipeline", "dlt_update", "dlt_maintenance"}}
+	records, err := costManager.GetResourcesCost(ctx, resources, startTime, endTime)
+	if err != nil {
+		return domain.AuditReport{}, err
+	}
+
+	// Aggregate per pipeline
+	type agg struct {
+		totalCost        float64
+		totalUsage       float64
+		updatesCount     int
+		maintenanceCount int
+		totalDurationSec float64
+		maxDurationSec   float64
+		currency         string
+	}
+	pipelines := map[string]*agg{}
+
+	for _, rec := range records {
+		id := rec.Resource.Name
+		if _, ok := pipelines[id]; !ok {
+			pipelines[id] = &agg{}
+		}
+		a := pipelines[id]
+		for _, c := range rec.Costs {
+			a.totalCost += c.TotalAmount
+			a.totalUsage += c.Value
+			if a.currency == "" {
+				a.currency = c.Currency
+			}
+		}
+		dur := rec.EndTime.Sub(rec.StartTime).Seconds()
+		a.totalDurationSec += dur
+		if dur > a.maxDurationSec {
+			a.maxDurationSec = dur
+		}
+		rt := rec.Resource.Service
+		if rt == "dlt_update" {
+			a.updatesCount++
+		}
+		if rt == "dlt_maintenance" {
+			a.maintenanceCount++
+		}
+	}
+
+	report := domain.AuditReport{
+		Workspace:    ws.Name,
+		ResourceType: "dlt_pipeline",
+		Period: domain.TimePeriod{
+			Start:    startTime,
+			End:      endTime,
+			Duration: int(endTime.Sub(startTime).Hours() / 24),
+		},
+		Summary:  map[string]any{},
+		Findings: []domain.AuditFinding{},
+	}
+
+	if len(records) == 0 {
+		report.Summary["no_activity"] = "No DLT usage records found in the selected period"
+		report.Findings = append(report.Findings, domain.AuditFinding{
+			Id:             "no_activity",
+			Resource:       domain.ResourceDef{Platform: "Databricks", Service: "dlt_pipeline", Name: "workspace"},
+			Issue:          "no_activity",
+			Description:    "No DLT pipeline activity detected in the selected time window.",
+			Recommendation: "Verify schedules/triggers and pipeline health. Consider reducing provisioned resources if unused.",
+			Severity:       domain.SeverityMedium,
+		})
+		return report, nil
+	}
+
+	var totalPipelines, highMaintenance, longRuns int
+	for pid, a := range pipelines {
+		totalPipelines++
+		maintenanceRatio := 0.0
+		totalEvents := a.updatesCount + a.maintenanceCount
+		if totalEvents > 0 {
+			maintenanceRatio = float64(a.maintenanceCount) / float64(totalEvents)
+		}
+		avgRunSec := 0.0
+		runEvents := a.updatesCount
+		if runEvents > 0 {
+			avgRunSec = a.totalDurationSec / float64(runEvents)
+		}
+
+		if maintenanceRatio > 0.3 && a.maintenanceCount >= 3 {
```

Comment:
NIT: I know it's not like this in the code base
but ideally this should be coming from some config
so cheap win/win is to move to ctor args in a form of settings but hardcode that config for now
then we can replace with proper DI with no side-effects